### PR TITLE
Account address

### PR DESCRIPTION
**Pull request by VelislavaC4Nexus**
### All pull requests:
* [ ] Have you considered security (e.g., XSS)?
* [ ] Have you considered desktop, mobile, and tablet form-factors?
* [ ] Have you considered [accessibility best practices](https://www.w3.org/WAI/standards-guidelines/wcag/)?

### Code
* [ ] Are the commits squashed into one commit?
* [ ] Is the [branch name and commit message](CONTRIBUTING.md#conventions-for-branch-names-and-commit-messages) following team convention?
* [ ] Is the code linted?
* [ ] Are all open issues, questions, and concerns by reviewers answered, resolved, and reviewed?

### Quality assurance
* [ ] If applicable, are unit tests written and are they passing?
* [ ] If applicable, are integration tests written and are they passing?
* [ ] Have checks linked in your pull request passed?

### Documentation
* [ ] Are there meaningful code comments included?

## Applicable to community contributors:
* [ ] Have you granted SFRA team access to your fork? [Grant access](CONTRIBUTING.md#community-contributors) 

### Documentation
* [ ] If applicable, is the change log updated?
* [ ] If applicable, have any UI text changes been reviewed by Documentation?
* [ ] If applicable, have any UI implementations been reviewed by the UX team?

### Security
* [ ] If applicable, has Security reviewed this code?
* [ ] If applicable, is a 3PP request submitted?

### CODE REVIEW

File: cartridges/app_custom_account_address/cartridge/client/default/js/checkout.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+'use strict';
+
+var processInclude = require('base/util');
+
+$(document).ready(function () {
+    console.log('checkout');
```
Comment (kalipiev) [inline cartridges/app_custom_account_address/cartridge/client/default/js/checkout.js]: Please remove this one

File: cartridges/app_custom_account_address/cartridge/client/default/js/checkout/shipping.js [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+
+var base = require("base/checkout/shipping");
+
+/**
+ * updates the shipping address form values within shipping forms
+ * @param {Object} shipping - the shipping (shipment model) model
+ */
+base.methods.updateShippingAddressFormValues = function (shipping) {
+    var addressObject = $.extend({}, shipping.shippingAddress);
+    console.log("addressObject" + addressObject);
```
Comment (kalipiev) [inline cartridges/app_custom_account_address/cartridge/client/default/js/checkout/shipping.js]: Please remove this one

File: cartridges/app_custom_account_address/cartridge/controllers/Account.js [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+ * @param {httpparameter} - dwfrm_profile_login_password - Input field for the shopper's password
+ * @param {httpparameter} - dwfrm_profile_login_passwordconfirm: - Input field for the shopper's password to confirm
+ * @param {httpparameter} - dwfrm_profile_customer_addtoemaillist - Checkbox for whether or not a shopper wants to be added to the mailing list
+ * @param {httpparameter} - csrf_token - hidden input field CSRF token
+ * @param {category} - sensitive
+ * @param {returns} - json
+ * @param {serverfunction} - post
+ */
+server.replace(
+    'SubmitRegistration',
```
Comment (kalipiev) [inline cartridges/app_custom_account_address/cartridge/controllers/Account.js:34]: Why do we need this replace ?

File: cartridges/app_custom_account_address/cartridge/controllers/Checkout.js [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+ * @memberof Checkout
+ * @param {middleware} - server.middleware.https
+ * @param {middleware} - consentTracking.consent
+ * @param {middleware} - csrfProtection.generateToken
+ * @param {querystringparameter} - stage - a flag indicates the checkout stage
+ * @param {category} - sensitive
+ * @param {renders} - isml
+ * @param {serverfunction} - get
+ */
+server.replace(
```
Comment (kalipiev) [inline cartridges/app_custom_account_address/cartridge/controllers/Checkout.js:31]: Same here ?

Comment (VelislavaC4Nexus) [reply cartridges/app_custom_account_address/cartridge/controllers/Checkout.js:31]: Он the 73rd line i remove checking "if (!shipment.shippingAddress)" to execute the code from COHelpers, so i need to replace it.

File: cartridges/app_custom_account_address/cartridge/controllers/Account.js (hunk 3 above)
Comment (VelislavaC4Nexus) [reply cartridges/app_custom_account_address/cartridge/controllers/Account.js:34]: Here in the form i had to submit data only for email and password.

File: cartridges/app_custom_account_address/cartridge/controllers/ReCaptcha.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    if (!token) {
+        res.json({
+            success: false,
+            errorMessage: Resource.msg('recaptcha.error.token', 'recaptcha', null)
+        });
+
+        return next();
+    }
+
+    const reCaptchaConfiguration = res.getViewData().reCaptchaConfig;
```
Comment (kalipiev) [inline cartridges/app_custom_account_address/cartridge/controllers/ReCaptcha.js]: Please move the logic below into a helper function

File: cartridges/app_custom_account_address/cartridge/models/account.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+/**
+ * Account class that represents the current customer's profile dashboard
+ * @param {Object} currentCustomer - Current customer
+ * @param {Object} addressModel - The current customer's preferred address
+ * @param {Object} orderModel - The current customer's order history
+ * @constructor
+ */
+function account(currentCustomer, addressModel, orderModel) {
+    module.superModule.call(this, currentCustomer, addressModel, orderModel);
```
Comment (kalipiev) [inline cartridges/app_custom_account_address/cartridge/models/account.js]: Please assign module.superModule to a variable at the top, like var base = module.superModule;

File: cartridges/app_custom_account_address/cartridge/models/account.js [hunk 7]
```diff
@@ -0,0 +1,49 @@
+'use strict';
```
Comment (kalipiev) [inline cartridges/app_custom_account_address/cartridge/models/account.js:1]: Do we need this extension, isn't it ok to have just the address model extended ?

File: cartridges/app_custom_account_address/cartridge/scripts/checkout/shippingHelpers.js [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+            lastName: req.form.lastName,
+            address1: req.form.address1,
+            address2: req.form.address2,
+            city: req.form.city,
+            stateCode: req.form.stateCode,
+            postalCode: req.form.postalCode,
+            countryCode: req.form.countryCode,
+            phone: req.form.phone,
+            companyName: req.form.companyName,
+           vat: req.form.vat,
```
Comment (kalipiev) [inline cartridges/app_custom_account_address/cartridge/scripts/checkout/shippingHelpers.js]: indendation

File: cartridges/app_custom_account_address/cartridge/scripts/checkout/shippingHelpers.js [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+            lastName: req.form.lastName,
+            address1: req.form.address1,
+            address2: req.form.address2,
+            city: req.form.city,
+            stateCode: req.form.stateCode,
+            postalCode: req.form.postalCode,
+            countryCode: req.form.countryCode,
+            phone: req.form.phone,
+            companyName: '',
+             vat: '',
```
Comment (kalipiev) [inline cartridges/app_custom_account_address/cartridge/scripts/checkout/shippingHelpers.js]: indentation

File: cartridges/app_custom_account_address/cartridge/static/default/css/empty.css
Comment (kalipiev) [inline cartridges/app_custom_account_address/cartridge/static/default/css/empty.css:1]: Do not commit static files

File: site_template/system-object-types-extensions.xml
Comment (kalipiev) [inline site_template/system-object-types-extensions.xml:1]: Please check the formatting of this XML file. The format should look like OrderAddress, though I can see incorrect indentation there.

File: cartridges/app_custom_account_address/cartridge/static/default/css/empty.css
Comment (VelislavaC4Nexus) [reply cartridges/app_custom_account_address/cartridge/static/default/css/empty.css:1]: i have typo in my gitignor file, will fix it

File: cartridges/app_custom_account_address/cartridge/models/account.js (hunk 7 above)
Comment (VelislavaC4Nexus) [reply cartridges/app_custom_account_address/cartridge/models/account.js:1]: Here in account. js i call new AddressModel with addressBook.addresses[i].raw to get the new custom attribute from address.row

File: cartridges/app_custom_account_address/cartridge/client/default/js/checkout.js (hunk 1 above)
Comment (VelislavaC4Nexus) [reply cartridges/app_custom_account_address/cartridge/client/default/js/checkout.js]: removed

File: cartridges/app_custom_account_address/cartridge/client/default/js/checkout/shipping.js (hunk 2 above)
Comment (VelislavaC4Nexus) [reply cartridges/app_custom_account_address/cartridge/client/default/js/checkout/shipping.js]: removed

File: cartridges/app_custom_account_address/cartridge/controllers/ReCaptcha.js (hunk 5 above)
Comment (VelislavaC4Nexus) [reply cartridges/app_custom_account_address/cartridge/controllers/ReCaptcha.js]: moved into helper function

File: cartridges/app_custom_account_address/cartridge/models/account.js (hunk 6 above)
Comment (VelislavaC4Nexus) [reply cartridges/app_custom_account_address/cartridge/models/account.js]: fixed

File: cartridges/app_custom_account_address/cartridge/scripts/checkout/shippingHelpers.js (hunk 8 above)
Comment (VelislavaC4Nexus) [reply cartridges/app_custom_account_address/cartridge/scripts/checkout/shippingHelpers.js]: fixed

File: cartridges/app_custom_account_address/cartridge/scripts/checkout/shippingHelpers.js (hunk 9 above)
Comment (VelislavaC4Nexus) [reply cartridges/app_custom_account_address/cartridge/scripts/checkout/shippingHelpers.js]: fixed

File: cartridges/app_custom_account_address/cartridge/client/default/js/addressFormRadioBtn.js [hunk 10]
```diff
@@ -0,0 +1,12 @@
+$(document).ready(function () {
+    $('.company-name-vat').hide();
+    $('input[type="radio"]').change(function () {
+        if ($(this).val() === 'Business Address') {
```
Comment (kalipiev) [inline cartridges/app_custom_account_address/cartridge/client/default/js/addressFormRadioBtn.js]: Please use a constant defined at the top of the module for "Business Address"

File: cartridges/app_custom_account_address/cartridge/controllers/ReCaptcha.js [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+            success: false,
+            errorMessage: Resource.msg('recaptcha.error.token', 'recaptcha', null)
+        });
+
+        return next();
+    }
+
+    const reCaptchaConfiguration = res.getViewData(token, reCaptchaConfiguration).reCaptchaConfig;
+
+    const reCaptchaResponse=reCaptchaHelpers.validateReCaptcha(token,reCaptchaConfiguration)
```
Comment (kalipiev) [inline cartridges/app_custom_account_address/cartridge/controllers/ReCaptcha.js]: Formatting - missing spaces around "="

**Pull request merged**